lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1207,6 +1185,38 @@ void WhenOp::build(OpBuilder &builder, OperationState &result, Value condition,
 // Expressions
 //===----------------------------------------------------------------------===//
 
+/// Type inference adaptor that narrows from the very generic MLIR
+/// `InferTypeOpInterface` to what we need in the FIRRTL dialect.
```

Comment:
Which is what?

Reply:
Just the operands and attrs, no context and regions. Also, we only ever produce one type, so we don't need the `SmallVectorImpl<Type> &resultTypes` to push multiple types into. I'll update the comment.